it/src/main/java/com/google/cloud/teleport/it/spanner/DefaultSpannerResourceManager.java

```diff
@@ -0,0 +1,219 @@
+/*
+ * Copyright (C) 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.it.spanner;
+
+import static com.google.common.hash.Hashing.goodFastHash;
+
+import com.google.cloud.spanner.DatabaseAdminClient;
+import com.google.cloud.spanner.DatabaseClient;
+import com.google.cloud.spanner.DatabaseId;
+import com.google.cloud.spanner.InstanceAdminClient;
+import com.google.cloud.spanner.InstanceConfigId;
+import com.google.cloud.spanner.InstanceId;
+import com.google.cloud.spanner.InstanceInfo;
+import com.google.cloud.spanner.KeySet;
+import com.google.cloud.spanner.Mutation;
+import com.google.cloud.spanner.ResultSet;
+import com.google.cloud.spanner.Spanner;
+import com.google.cloud.spanner.SpannerOptions;
+import com.google.cloud.teleport.it.spanner.SpannerResourceManagerException.CreateDatabaseException;
+import com.google.cloud.teleport.it.spanner.SpannerResourceManagerException.CreateInstanceException;
+import com.google.cloud.teleport.it.spanner.SpannerResourceManagerException.UpdateDatabaseException;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.CharMatcher;
+import com.google.common.collect.ImmutableList;
+import com.google.common.hash.HashFunction;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.util.concurrent.ExecutionException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Default class for implementation of {@link SpannerResourceManager} interface.
+ *
+ * <p>There are one instance, one database, and multiple tables per manager object. The tables are
```

Comment:
Few comments on this paragraph:

1. I would rephrase that it it supports multiple tables, not that it necessarily will have multiple tables.
2. The comment on how instances, databases, and tables are structured is unnecessary. That's standard Spanner details. The reason for mentioning the number is so that users know what it can and can't be used for in a test.
3. Avoid giving implementation details. The instance being created when the object is is relevant, but we don't have to  specifically call out `build()` as the place that it happens, and we definitely shouldn't discuss *how* it's doing that. (Besides, `createInstance` isn't public.) Doc comments are about describing behavior, not implementation.
4. The comment on database creation should be added to the `createTable` method. Specifically, it only does so on the first call to `createTable`, so the state impact of later calls is different.